### ISSUE TITLE
firrtl.node type verification

### ISSUE THREAD

**Issue by darthscsi** [llvm/circt#395]
The following now fails

```
module  {
  firrtl.circuit "foo" {
    firrtl.extmodule @bar(%baz: !firrtl.uint<1>)
    firrtl.module @foo() {
      %fetch = firrtl.instance @bar {name = "fetch"} : !firrtl.bundle<baz: flip<uint<1>>>
      %1454 = firrtl.subfield %fetch("baz") : (!firrtl.bundle<baz: flip<uint<1>>>) -> !firrtl.flip<uint<1>>
      %hits_1_7 = firrtl.node %1454 {name = "hits_1_7"} : !firrtl.flip<uint<1>>
    }
  }
}
```

**Comment by seldridge** [comment]
This may be fixed with #381. Relevant code: https://github.com/llvm/circt/pull/381/files#diff-ee090cee6cf72cf93f7f3c6685d8a29e8d5b79fee0e3d68691ba10f32c1da42aR2118

**Comment by seldridge** [comment]
I think there's a couple of things going on here:

1. The FIRRTL spec requires that nodes are passive. We should add a verifier to enforce this so that this fails earlier than during conversion to the RTL dialect.
2. CHIRRTL (and the SFC) accept flipped nodes that are passive under the flip, e.g., you can have a node pointing at an output port of the current module.

(1) should be disallowed via verification of the node op. I'll look into adding a verifier for this.

(2) should be fixed up via #381 where a node passive under a single outer flip is converted to passive via an `asPassive` operand. #381 implements this.
